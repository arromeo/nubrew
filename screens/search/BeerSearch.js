import React from 'react';
import { StyleSheet, View, Text, FlatList } from 'react-native';

export default class BeerSearch extends React.Component {
  render() {
    const styles = this.props.styles;
    return (
      <View>
        <FlatList
        data={this.props.data}
        keyExtractor={item => item.beer_id.toString()}
        renderItem={({item}) => 
          <View style={styles.listItemContainer}>
            <View style={styles.searchResultContainer}>
              <Text>Brewery: {item.brewery_name}'s</Text>
              <Text>Beer Name: {item.beer_name}</Text>
              <Text>Type: {item.category}</Text>
              <Text>{item.beer_description}</Text>
              <Text>IBU: {item.ibu} - ABV: {item.abv}</Text>
            </View>
          </View>
        }
        />
      </View>
    )
  }
}
