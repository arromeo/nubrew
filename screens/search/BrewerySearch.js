import React from 'react';
import { ScrollView, StyleSheet, Image, TouchableOpacity, View, Text, FlatList } from 'react-native';

export default class BrewerySearch extends React.Component {

  render() {
    console.log(this.props.data);
    return (
        <ScrollView style={styles.container}>
          <FlatList
            data={this.props.data}
            keyExtractor={item => item.id.toString()}
            renderItem={({item}) => 
              <TouchableOpacity
                style={[styles.eventContainer, styles.homeScreenFilename]}
                onPress={() => {
                  this.props.navigate('Detail', {
                    id: item.id,
                    category: "Brewery"
                  });
              }}>
              <View style={[styles.eventDetailsContainer, styles.homeScreenFilename]}>
                <View>
                  <Image
                    style={{width: 75, height: 75}}
                    source={{uri: item.img_url }}
                  />
                </View>
                <View>
                  <Text style={styles.eventTitle}>{item.name}</Text>
                  <Text style={styles.eventDetails}>{item.street_address}</Text>
                  <Text style={styles.eventDetails}>{item.city}, {item.province}</Text>
                  <Text style={styles.eventDetails}>{item.postal_code}</Text>
                </View>
              </View>
            </TouchableOpacity>
          }/>
        </ScrollView>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 15,
    backgroundColor: '#fff',
  },
  eventDetailsContainer: {
    flexDirection: "row",
    flex: 1,
    margin: 5,
  },
  eventDetails: {
    marginLeft: 7,
    justifyContent: "center",
    fontSize: 11
  },
  eventTitle: {
    marginLeft: 5,
    fontWeight: "bold"
  }
});