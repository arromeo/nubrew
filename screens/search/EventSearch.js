import React from 'react';
import { TouchableOpacity, View, Text, FlatList } from 'react-native';

export default class EventSearch extends React.Component {
  render() {
    const styles = this.props.styles;
    return (
        <FlatList
        data={this.props.data}
        keyExtractor={item => item.id.toString()}
        renderItem={({item}) => 
          <TouchableOpacity style={styles.listItemContainer}
            onPress={() => {
              this.props.navigate('Detail', {
                category: "Event",
                id: item.id,
              })}
            }
          >
            <View style={styles.searchResultContainer}>
              <Text>{item.name}</Text>
              <Text>{item.time}</Text>
              <Text>{item.details}</Text>
            </View>
          </TouchableOpacity>
        }
        />
    )
  }
}
