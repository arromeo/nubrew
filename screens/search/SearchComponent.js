import React from 'react';
import { View, Picker, StyleSheet } from 'react-native';
import { SearchBar } from 'react-native-elements';

export default class SearchComponent extends React.Component {
  render() {
    return (
      <View>
        <SearchBar
          showLoading
          lightTheme
          ref={search => this.search = search}
          onChangeText={(event) => this.props.changeInput(event)}
          value={this.props.input}
          placeholder='Type Here...' />
        <View style={styles.pickerValueContainer}>
          <Picker
            selectedValue={this.props.pickerValue}
            style={styles.pickerValueContainer}
            onValueChange={(event) => this.props.pickCategory(event)}>
            <Picker.Item label="Beers" value="Beer" />
            <Picker.Item label="Breweries" value="Brewery" />
            <Picker.Item label="Stores" value="Store" />
            <Picker.Item label="Events" value="Event" />
          </Picker>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  pickerValueContainer: {
    flex: 0.5,
    width: '100%',
    justifyContent: 'center', 
    alignItems: 'center',
  },
});

