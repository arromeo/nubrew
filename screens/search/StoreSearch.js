import React from 'react';
import { ScrollView, StyleSheet, Image, TouchableOpacity, View, Text, FlatList } from 'react-native';

export default class EventSearch extends React.Component {
  render() {
    return (
        <ScrollView style={styles.container}>
            <View>
              <FlatList
                data={this.props.data}
                keyExtractor={item => item.id.toString()}
                renderItem={({item}) => 
                <TouchableOpacity
                  style={[styles.eventContainer, styles.homeScreenFilename]}
                  onPress={() => {
                    this.props.navigate('Detail', {
                      id: item.id,
                      category: "Store"
                    });
                }}>
                  <View style={[styles.eventDetailsContainer, styles.homeScreenFilename]}>
                  <View>
                    <Image
                      style={{width: 75, height: 75}}
                      source={{uri: item.img_url }}
                    />
                  </View>
                  <View>
                    <Text style={styles.eventTitle}>{item.name}</Text>
                    <Text style={styles.eventDetails}>{item.street_address}</Text>
                    <Text style={styles.eventDetails}>{item.city}, {item.province}</Text>
                    <Text style={styles.eventDetails}>{item.postal_code}</Text>
                  </View>
                </View>
              </TouchableOpacity>
            }/>
          </View>
        </ScrollView>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 15,
    backgroundColor: '#fff',
  },
  headerFont: {
    color: 'black',
    fontWeight: 'bold',
    marginBottom: 20,
    flex: 1,
    textAlign: 'center',
    justifyContent: 'center',
  },
  contentContainer: {
    paddingBottom: 5,
    flexDirection: "row",
    justifyContent: "center",
    alignItems: 'flex-start',
  },
  eventDetailsContainer: {
    flexDirection: "row",
    flex: 1,
    margin: 5,
  },
  eventDetails: {
    marginLeft: 7,
    justifyContent: "center",
    fontSize: 11
  },
  eventTitle: {
    marginLeft: 5,
    fontWeight: "bold"
  }
});