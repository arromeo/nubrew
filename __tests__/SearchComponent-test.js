import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import '../__mock__/xhr-mock';

import BeerSearch from '../screens/search/BeerSearch';
import BrewerySearch from '../screens/search/BrewerySearch';
import EventSearch from '../screens/search/EventSearch';
import StoreSearch from '../screens/search/StoreSearch';
import SearchComponent from '../screens/search/SearchComponent';

import props from '../__mock__/props';

describe('Search components render appropriately', () => {
  it('renders the BeerSearch component', () => {
    const tree = renderer.create(<BeerSearch navigate={props.navigation.navigate} data={props.data}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders the BrewerySearch component', () => {
    const tree = renderer.create(<BrewerySearch navigate={props.navigation.navigate} data={props.data}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders the EventSearch component', () => {
    const tree = renderer.create(<EventSearch navigate={props.navigation.navigate} data={props.data}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders the StoreSearch component', () => {
    const tree = renderer.create(<StoreSearch navigate={props.navigation.navigate} data={props.data}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders the SearchComponent component', () => {
    const tree = renderer.create(<SearchComponent/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
})