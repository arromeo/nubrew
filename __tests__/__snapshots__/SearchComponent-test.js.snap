// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search components render appropriately renders the BeerSearch component 1`] = `null`;

exports[`Search components render appropriately renders the BrewerySearch component 1`] = `null`;

exports[`Search components render appropriately renders the EventSearch component 1`] = `null`;

exports[`Search components render appropriately renders the SearchComponent component 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#393e42",
          "borderBottomColor": "#000",
          "borderBottomWidth": 1,
          "borderTopColor": "#000",
          "borderTopWidth": 1,
        },
        Object {
          "backgroundColor": "#e1e8ee",
          "borderBottomColor": "#e1e1e1",
          "borderTopColor": "#e1e1e1",
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Type Here..."
      placeholderTextColor="#86939e"
      showLoading={true}
      style={
        Array [
          Object {
            "backgroundColor": "#303337",
            "borderRadius": 3,
            "color": "#86939e",
            "fontSize": 17.5,
            "height": 30,
            "margin": 8,
            "overflow": "hidden",
            "paddingLeft": 26,
            "paddingRight": 19,
          },
          Object {
            "backgroundColor": "#bdc6cf",
          },
          false,
          false,
          undefined,
          undefined,
          false,
        ]
      }
      underlineColorAndroid="transparent"
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#86939e",
            "fontSize": 16,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "position": "absolute",
              "top": 15.5,
            },
            Object {
              "left": 16,
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      î¢¶
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.5,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "Beers",
              "textColor": undefined,
              "value": "Beer",
            },
            Object {
              "label": "Breweries",
              "textColor": undefined,
              "value": "Brewery",
            },
            Object {
              "label": "Stores",
              "textColor": undefined,
              "value": "Store",
            },
            Object {
              "label": "Events",
              "textColor": undefined,
              "value": "Event",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Search components render appropriately renders the StoreSearch component 1`] = `null`;
